### PR TITLE
Refactor index js

### PR DESCRIPTION
**Pull request by jocalafe**
# Description

Moved all zones data to a separate file for each of the providers and created an individual "spider" script for each provider.

Fixes # #27 

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by oscard0m** [comment]
Branch checkouted an working. 

## Tests done:

```bash
npm start -- --fotocasa
````

**Output**:
```bash
> phantomjs --ssl-protocol=any --ignore-ssl-errors=true index.js "--fotocasa"


-------------------------------------------------------------------
              FOTOCASA
-------------------------------------------------------------------

ReferenceError: Can't find variable: Set

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              CIUTAT VELLA
-------------------------------------------------------------------

Piso en Riereta, 3
1.050 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/calefaccion-amueblado-riereta-3-147273293?RowGrid=1&tti=3&opi=300


Apartamento en Gótico
1.250 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/calefaccion-amueblado-barri-gotic-147318633?RowGrid=2&tti=3&opi=300


Apartamento en Escudellers Blancs
1.200 €/mes
hace 18 minutos
https://www.fotocasa.es/vivienda/barcelona-capital/aire-acondicionado-calefaccion-amueblado-television-internet-escudellers-blancs-147614890?RowGrid=3&tti=3&opi=300


Apartamento en Gótico
1.250 €/mes
hace 15 horas
https://www.fotocasa.es/vivienda/barcelona-capital/calefaccion-amueblado-barri-gotic-147318633?RowGrid=4&tti=3&opi=300


Piso en Parallel, 70
1.200 €/mes
hace 17 horas
https://www.fotocasa.es/vivienda/barcelona-capital/aire-acondicionado-ascensor-amueblado-parallel-70-147603784?RowGrid=5&tti=3&opi=300


Piso en Carlos Tercero, 41
1.250 €/mes
hace 22 horas
https://www.fotocasa.es/vivienda/barcelona-capital/calefaccion-terraza-trastero-ascensor-amueblado-se-aceptan-mascotas-no-amueblado-carlos-tercero-41-147599386?RowGrid=6&tti=3&opi=300


Piso en Pi I Molist
1.050 €/mes
hace 23 horas
https://www.fotocasa.es/vivienda/barcelona-capital/aire-acondicionado-ascensor-amueblado-television-pi-i-molist-147598277?RowGrid=7&tti=3&opi=300


ReferenceError: Can't find variable: Set

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              EIXAMPLE
-------------------------------------------------------------------

Piso en Provença, 198
1.200 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/aire-acondicionado-calefaccion-ascensor-amueblado-television-provenca-198-147301510?RowGrid=1&tti=3&opi=300


Piso en Tamarit
1.100 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/aire-acondicionado-calefaccion-ascensor-patio-amueblado-tamarit-146258151?RowGrid=2&tti=3&opi=300


Piso en Arago, 343
1.250 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/calefaccion-ascensor-amueblado-arago-343-147560379?RowGrid=3&tti=3&opi=300


Piso en Av Gaudí
1.200 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/ascensor-amueblado-television-av-gaudi-147564611?RowGrid=4&tti=3&opi=300


Piso en Villarroel
1.050 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/ascensor-amueblado-villarroel-147565341?RowGrid=5&tti=3&opi=300


Piso en Villarroel
1.050 €/mes
hace 17 horas
https://www.fotocasa.es/vivienda/barcelona-capital/calefaccion-ascensor-amueblado-villarroel-147603571?RowGrid=6&tti=3&opi=300


Piso en Eixample - Fort Pienc
1.100 €/mes
hace 21 horas
https://www.fotocasa.es/vivienda/barcelona-capital/ascensor-amueblado-television-sagrada-familia-147559042?RowGrid=7&tti=3&opi=300


Piso en Eixample - Fort Pienc
1.080 €/mes
hace 21 horas
https://www.fotocasa.es/vivienda/barcelona-capital/aire-acondicionado-calefaccion-ascensor-amueblado-television-se-aceptan-mascotas-sagrada-familia-147510504?RowGrid=8&tti=3&opi=300


ReferenceError: Can't find variable: Set

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              GRACIA
-------------------------------------------------------------------

Piso en Vila de Gràcia
1.150 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/terraza-ascensor-amueblado-vila-de-gracia-146575090?RowGrid=1&tti=3&opi=300


Piso en Travesera de Gracia, 102
1.150 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/ascensor-amueblado-travesera-de-gracia-102-147232819?RowGrid=2&tti=3&opi=300


Piso en Martinez de la Rosa
975 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/calefaccion-terraza-amueblado-martinez-de-la-rosa-147479526?RowGrid=3&tti=3&opi=300


Piso en Escorial, 17
1.000 €/mes
hace 12 horas
https://www.fotocasa.es/vivienda/barcelona-capital/aire-acondicionado-terraza-ascensor-amueblado-internet-escorial-17-147609768?RowGrid=4&tti=3&opi=300


ReferenceError: Can't find variable: Set

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              LES CORTS
-------------------------------------------------------------------

Piso en Horta - Guinardó - El Guinardó
1.100 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/amueblado-hostafrancs-147566943?RowGrid=1&tti=3&opi=300


Piso en Corral
1.100 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/terraza-ascensor-amueblado-television-corral-147501386?RowGrid=2&tti=3&opi=300


Piso en Corral
1.100 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/ascensor-amueblado-corral-147501394?RowGrid=3&tti=3&opi=300


Piso en Corral
1.100 €/mes
hace 15 horas
https://www.fotocasa.es/vivienda/barcelona-capital/terraza-ascensor-amueblado-television-corral-147501386?RowGrid=4&tti=3&opi=300


Piso en Corral
1.100 €/mes
hace 15 horas
https://www.fotocasa.es/vivienda/barcelona-capital/ascensor-amueblado-corral-147501394?RowGrid=5&tti=3&opi=300


Piso en Sants - Montjuïc - Sants
1.200 €/mes
hace 22 horas
https://www.fotocasa.es/vivienda/barcelona-capital/aire-acondicionado-calefaccion-amueblado-television-sants-147599796?RowGrid=6&tti=3&opi=300


ReferenceError: Can't find variable: Set

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              SANTS
-------------------------------------------------------------------

Piso en Poblet, 6
1.200 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/calefaccion-trastero-ascensor-amueblado-poblet-6-147584920?RowGrid=1&tti=3&opi=300


Piso en Europa
1.200 €/mes
hace 14 horas
https://www.fotocasa.es/vivienda/barcelona-capital/zona-comunitaria-ascensor-amueblado-europa-147536784?RowGrid=2&tti=3&opi=300


Piso en Pisuerga
1.050 €/mes
hace 16 horas
https://www.fotocasa.es/vivienda/barcelona-capital/terraza-ascensor-amueblado-pisuerga-147604656?RowGrid=3&tti=3&opi=300


Piso en Pintor Tapiro, 53
1.075 €/mes
hace 17 horas
https://www.fotocasa.es/vivienda/barcelona-capital/calefaccion-amueblado-se-aceptan-mascotas-pintor-tapiro-53-147604081?RowGrid=4&tti=3&opi=300
```

-----------------------------------

```bash
npm start
```

**Output**

```bash
> phantomjs --ssl-protocol=any --ignore-ssl-errors=true index.js


-------------------------------------------------------------------
              IDEALISTA
-------------------------------------------------------------------

TypeError: undefined is not an object (evaluating 's')

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              CIUTAT VELLA: No new apartments
-------------------------------------------------------------------

TypeError: undefined is not an object (evaluating 's')

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              EIXAMPLE: No new apartments
-------------------------------------------------------------------

TypeError: undefined is not an object (evaluating 's')

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              GRACIA: No new apartments
-------------------------------------------------------------------

TypeError: undefined is not an object (evaluating 's')

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              LES CORTS: No new apartments
-------------------------------------------------------------------

TypeError: undefined is not an object (evaluating 's')

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              SANTS: No new apartments
-------------------------------------------------------------------
```

This output with missing data to me is related to #25 

---------------------------------------------------------------------------------

```bash
npm start -- --fotocasa --les-corts
```

**Output**

```bash
> phantomjs --ssl-protocol=any --ignore-ssl-errors=true index.js "--fotocasa" "--les-corts"


-------------------------------------------------------------------
              FOTOCASA
-------------------------------------------------------------------

ReferenceError: Can't find variable: Set

  phantomjs://code/index.js:30 in onError
****Status: success****

[object Object]

-------------------------------------------------------------------
              LES CORTS
-------------------------------------------------------------------

Piso en Horta - Guinardó - El Guinardó
1.100 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/amueblado-hostafrancs-147566943?RowGrid=1&tti=3&opi=300


Piso en Corral
1.100 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/terraza-ascensor-amueblado-television-corral-147501386?RowGrid=2&tti=3&opi=300


Piso en Corral
1.100 €/mes
ahora
https://www.fotocasa.es/vivienda/barcelona-capital/ascensor-amueblado-corral-147501394?RowGrid=3&tti=3&opi=300


Piso en Corral
1.100 €/mes
hace 15 horas
https://www.fotocasa.es/vivienda/barcelona-capital/terraza-ascensor-amueblado-television-corral-147501386?RowGrid=4&tti=3&opi=300


Piso en Corral
1.100 €/mes
hace 15 horas
https://www.fotocasa.es/vivienda/barcelona-capital/ascensor-amueblado-corral-147501394?RowGrid=5&tti=3&opi=300


Piso en Sants - Montjuïc - Sants
1.200 €/mes
hace 22 horas
https://www.fotocasa.es/vivienda/barcelona-capital/aire-acondicionado-calefaccion-amueblado-television-sants-147599796?RowGrid=6&tti=3&opi=300
```

File: index.js [hunk 1]
```diff
@@ -73,9 +32,10 @@ page.onError = function(msg, trace) {
 
 function scrapPage(status) {
     console.log("****Status: " + status + "****\n");
+    console.log(fotocasaSpider);
```
Comment (oscard0m) [inline index.js]: I think we should remove this `console.log`

**Review by oscard0m — CHANGES_REQUESTED** [review]
Check small comment :)

**Comment by oscard0m** [comment]
Adding @nerder and @juanchaur since this initial PR's are the foundations but to me it's completely fine and works :)

🚀

File: index.js (hunk 1 above)
Comment (jocalafe) [reply index.js]: woops haha se coló

Comment (oscard0m) [reply index.js]: Done

**Review by oscard0m — APPROVED** [review]
Removed console.log

**Pull request merged**